src/OrleansBondUtils/BondSerializer.cs

```diff
@@ -191,15 +197,35 @@ private void Register(Type type)
             var clonerType = typeof(Cloner<>);
             var realType = clonerType.MakeGenericType(type);
             var clonerInstance = Activator.CreateInstance(realType);
-            var cloneMethod = realType.GetMethod("Clone").MakeGenericMethod(type);
-            var copierDelegate = cloneMethod.CreateDelegate(
-                    typeof(Func<,>).MakeGenericType(new[] { type, type }),
-                    clonerInstance);
             var serializer = new BondTypeSerializer(type);
             var deserializer = new BondTypeDeserializer(type);
-            CopierDictionary.TryAdd(type.TypeHandle, copierDelegate);
+            var parameterExpression1 = Expression.Parameter(typeof(object));
```

Comment:
We should get perf tests, but the difference is in what type information is present: the old method used a `Delegate`, which is typically slower than the compiled expression because of safety checks (from my understanding).
See here for example: http://www.palmmedia.de/Blog/2012/2/4/reflection-vs-compiled-expressions-vs-delegates-performance-comparision